core/lib/zksync_core/src/reorg_detector/mod.rs

```diff
@@ -156,9 +225,13 @@ impl ReorgDetector {
 
             // At this point we're certain that if we detect a reorg, it's real.
             tracing::trace!("Checking for reorgs - L1 batch #{sealed_l1_batch_number}");
-            if self.root_hashes_match(sealed_l1_batch_number).await? {
+            if self.root_hashes_match(sealed_l1_batch_number).await?
+                && self.miniblock_hashes_match(sealed_miniblock_number).await?
+            {
                 EN_METRICS.last_correct_batch[&CheckerComponent::ReorgDetector]
                     .set(sealed_l1_batch_number.0.into());
+                EN_METRICS.last_correct_miniblock[&CheckerComponent::ReorgDetector]
+                    .set(sealed_miniblock_number.0.into());
                 tokio::time::sleep(SLEEP_INTERVAL).await;
             } else {
                 tracing::warn!(
```

Comment:
I'd prefer to log more precise messages, i.e log whether L1 batch or miniblock hash is diverging